### ISSUE TITLE
Set expectations for GitHub Issue triage

### ISSUE THREAD

**Issue by chadwhitacre** [getsentry/open.sentry.io#1]
Coming here most immediately from https://github.com/getsentry/sentry-symfony/issues/436 ...

We need to systematize our approach to GitHub Issue triage and as part of that set some expectations with the community about process and timelines.

Private: https://app.asana.com/0/1199917695387960/1199915875658265

**Comment by chadwhitacre** [comment]
Here's a proposal for a basic process (issue = issue-or-PR):

1. **Route.** Within 72 hours of a new issue being opened, the Open Source team assigns a `Component: ___` label and @mentions the relevant team ([assign, eventually](https://github.com/dear-github/dear-github/issues/170)?).
1. **Triage.** Within 7 days of an issue being routed, assigned team closes with a comment, or labels with one of the following:
    - `Status: Needs More Information`—not actionable (stalebot will reap eventually)
    - `Status: On Hold`—real issue but not prioritized right now (stalebot will ignore)
    - `Status: In Progress`—real issue and actively being worked on (stalebot will ignore)
1. **Resolve.** Most teams manage their day-to-day workflow in private (Jira, Asana), but they are responsible to keep their public GitHub issues up-to-date.

**Comment by chadwhitacre** [comment]
This doesn't address outside collaborator involvement in triage/work, which was the main sticking point on https://github.com/getsentry/sentry-symfony/issues/436. 🤔

I do think every repo under the `getsentry` org should map to exactly one "routable" team, even if for some repos there is a larger outside contributor community than for others.

**Comment by mitsuhiko** [comment]
I think the core of the issue here is that we have an ill-defined process about dealing with customer issues coming in via GitHub. The regular triaging process is that support deals with incoming customer issues which are internally escalated to engineering if necessary. When that happens we are now creating a GitHub issue to track this.

Nothing stops a customer to create that GitHub issue directly obviously. However issues coming in that way are handled differently priority wise compared to issues coming in via customer support. For instance the symphony repository is an example for a repo that's not monitored much for customer requests by client infra as issues are unlikely to be stale there given how things are done currently.

So the core issue is: how do we deal with escalated customer support issues vs issues filed directly on GitHub :)

**Comment by chadwhitacre** [comment]
> The regular triaging process is that support deals with incoming customer issues which are internally escalated to engineering if necessary. When that happens we are now creating a GitHub issue to track this.

Is this real today? I've been expecting to work with @MistyED (and @manuzope @jonesphillip et al.) to make this real for Ecosystem-related issues in Q1.

> For instance the symphony repository is an example for a repo that's not monitored much for customer requests by client infra

If the right person/team is not monitoring the queue for `sentry-symfony` then that's a problem we need to fix. Outside contributors can be part of the solution for triage but imo we need to have paid eyes on all inbound issues with SLAs (_for triage_, not resolution) even if contributors are highly active in some repos. Something as simple as paid staff "officially" assigning https://github.com/getsentry/sentry-symfony/issues/436 to @Jean85, or at least 👍 one of his comments, or some such, might've nipped @iaicam's frustration in the bud. Even better will be to have a page up at https://open.sentry.io/processes/triage or some such that sets expectations that @Jean85 or whomever can link to when questions come up.

> So the core issue is: how do we deal with escalated customer support issues vs issues filed directly on GitHub :)

Where we're headed (with Ecosystem to begin with) is that escalated customer support issues end up on GitHub and feed into engineering teams' workflows (Jira, Asana, what-have-you) just as if the customer had created it directly on GitHub themselves. I think it is then each team's responsibility to stay on top of their inbound GitHub queue, prioritize work as they see fit, and communicate with customers via GitHub and whatever other channels make sense (Support, private Slack, email, etc.). The fact that an issue came in via escalated support and support created an issue on GitHub vs. direct creation on GitHub is another bit of signal for product owners to decide what to work on next.

**Comment by mitsuhiko** [comment]
> Is this real today? I've been expecting to work with @MistyED (and @manuzope @jonesphillip et al.) to make this real for Ecosystem-related issues in Q1.

It's becoming real for all some platform SDKs at least. It used to be tracked in Asana but mobile and web for instance already moved to github except for when the issue references customer information.

> If the right person/team is not monitoring the queue for sentry-symfony then that's a problem we need to fix.

It's not like the repo is not monitored at all. For instance when I saw the initial reply (on Monday) to the issue by Jean85 I was sufficiently satisfied with the response and did not engage more. Before today it never came up that contributors would not be self sufficient in replying to issues. We generally trust the folks we give access to. I also for what it's worth do not agree with the idea that we need to demote contributors from their status as a result of this response.

> Where we're headed (with Ecosystem to begin with) is that escalated customer support issues end up on GitHub and feed into engineering teams' workflows (Jira, Asana, what-have-you) just as if the customer had created it directly on GitHub themselves.

That cannot work because github is ill-equipped for this purpose. A lot of issues posted on github by customers are inferior to issues provided to engineering via the support triage flow for a handful of reasons:

- issues reported via the support issue are associated with the customer account so it's easier to figure out details about the issue before it comes towards SDK engineering for instance
- there are many times the issues coming in than can be worked with and we're definitely prioritizing issues from paying accounts over issues from free users. Determining this from GitHub issues alone is impossible.

I do not see a way in which the customer's experience will be of equal quality creating a GitHub issue vs going through support. We're tracking customer response times as a business metric for everything that comes in via zendesk and we also track escalations. Stuff that appears on GitHub goes straight towards engineering and will have to compete with internally escalated issues. I absolutely believe that GitHub issues should be handled with the necessary priority but we have to figure out a way to guide the issue creation away from GitHub towards support first otherwise what's the point of having triage? In particular for cases like the one reported in the original issue it's a case of customer support problems and not even a bug report from what I can tell.

**Comment by Jean85** [comment]
Thank you both @mitsuhiko & @chadwhitacre for taking over and handling this issue, also here in the open.

> Before today it never came up that contributors would not be self sufficient in replying to issues. We generally trust the folks we give access to. I also for what it's worth do not agree with the idea that we need to demote contributors from their status as a result of this response.

I agree, and I understand that if a contributor is active you (Sentry paid employees) do not need to do much. But getsentry/sentry-symfony#436 highlighted that the moment an user says "but I expect a paid-service level of support" it's out of our (contributors) hands. So, we need to know:
 * that the repos are watched (so I don't have any obligation to answer if I'm not sure how or I'm unavailable)
 * what to do in those cases where the client would need to be escalated to support (who I need to ping? how?); as of now I just pinged @HazAT on Discord, but that was it; it's not a reliable escalation mechanism

So, I also agree with @chadwhitacre, the triaging SLA & process needs to be in place & clarified.

Pinging @ste93cry & @stayallive also, since they're in the same boat as me regarding PHP-related repos.

**Comment by chadwhitacre** [comment]
> that the repos are watched (so I don't have any obligation to answer if I'm not sure how or I'm unavailable)

We've defined and published [a process for handling inbound issues and PRs](https://develop.sentry.dev/processing-tickets/), which includes a reference to external triagers. We're working on habituating this process with teams working on `sentry`, `sentry-docs`, and `onpremise` this quarter, and hopefully in Q2 we can work on taking into account repos in the long tail. Heads up that later in Q1 we expect this doc to migrate to https://open.sentry.io/, which currently clones a page from the marketing site but which we're rebuilding from scratch.

> what to do in those cases where the client would need to be escalated to support (who I need to ping? how?); as of now I just pinged @HazAT on Discord, but that was it; it's not a reliable escalation mechanism

I've captured this on https://github.com/getsentry/open.sentry.io/issues/3 to address with the new site.

> I do not see a way in which the customer's experience will be of equal quality creating a GitHub issue vs going through support.

I think the new process helps. What if we had a welcome message for valid issues to communicate the 1-week triage SLA and nudge paid customers towards support?
